ingress-node-firewall/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -0,0 +1,169 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols
+type IngressNodeFirewallICMPRule struct {
+	// ICMPType define ICMP Type Numbers (RFC 792).
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum:=255
+	// +kubebuilder:validation:Minimum:=0
+	ICMPType uint8 `json:"icmpType,omitempty"`
+
+	// ICMPCode define ICMP Code ID (RFC 792).
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum:=255
+	// +kubebuilder:validation:Minimum:=0
+	ICMPCode uint8 `json:"icmpCode,omitempty"`
+}
+
+// IngressNodeFirewallProtoRule define ingress node firewall rule for TCP, UDP and SCTP protocols
+type IngressNodeFirewallProtoRule struct {
+	// Ports can be a single port or range of ports using start-end range format
+	// if admin wanted to filter single port then setting a single port as an integer for example Ports: 80
+	// if admin wanted to filter range of ports then using "start-end" range string format for example Ports: "80-100"
+	//+kubebuilder:validation:Optional
+	Ports intstr.IntOrString `json:"ports,omitempty"`
+}
+
+// IngressNodeProtocolConfig is a discriminated union of protocol's specific configuration.
+// +union
+type IngressNodeProtocolConfig struct {
+	// ProtocolRule define ingress node firewall rule for TCP, UDP and SCTP protocols.
+	// +kubebuilder:validation:Optional
+	ProtocolRule *IngressNodeFirewallProtoRule `json:"protoRule,omitempty"`
+
+	// ICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols.
+	// +kubebuilder:validation:Optional
+	ICMPRule *IngressNodeFirewallICMPRule `json:"icmpRule,omitempty"`
+}
+
+// IngressNodeFirewallProtocolRule define ingress node firewall rule per protocol
+type IngressNodeFirewallProtocolRule struct {
+	// Order define order of execution of ingress firewall rules .
+	// + index 0 is used internally as catch all for unclassified packets matching the same sourceCIDR.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum:=1
+	Order uint32 `json:"order"`
+
+	// ProtocolConfig is a discriminated union of protocol's specific configuration.
+	// if not specificed matching will be based on protocol value and the protocol configs like dstPort/type/code will be ignored
+	// +kubebuilder:validation:Optional
+	ProtocolConfig IngressNodeProtocolConfig `json:"protocolConfig"`
+
+	// Protocol can be ICMP, ICMPv6, TCP, SCTP or UDP.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="ICMP";"ICMPv6";"TCP";"UDP";"SCTP"
+	Protocol IngressNodeFirewallRuleProtocolType `json:"protocol"`
+
+	// Action can be Allow or Deny, default action is deny
```

Comment:
Use the json tag name and complete sentences, remember these comments form part of the user documentation
```suggestion
	// action can be Allow or Deny, default action is Deny.
```